idle_agent.py

```diff
@@ -0,0 +1,83 @@
+import subprocess
+import os
+import boto3
+import datetime
+import instance_idle_pb2
+import time
+import base64
+import asyncio
+
+AWS_ACCESS_KEY_ID = os.environ["AWS_ACCESS_KEY_ID"]
+AWS_SECRET_ACCESS_KEY = os.environ["AWS_SECRET_ACCESS_KEY"]
+REGION_NAME = os.environ["REGION_NAME"]
+QUEUE_NAME = os.environ["QUEUE_NAME"]
+CHECK_INTERVAL = int(os.environ["CHECK_INTERVAL"])
+PORT = os.environ["PORT"]
+IDLE_TIME = int(os.environ["IDLE_TIME"])
+INSTANCE_ID = os.environ["INSTANCE_ID"]
+
+
+def count_connections():
+    result = subprocess.run(
+        ["ss", "-t", "-H", "-n", f"dport = {PORT}"], capture_output=True
+    )
+    return len(result.stdout.decode("utf-8").strip().split("\n")) - 1
+
+
+def get_last_connection_timestamp():
+    result = subprocess.run(
+        ["ss", "-t", "-H", "-n", f"dport = {PORT}"], capture_output=True
+    )
```

Comment:
This can be made asynchronous as well:
https://docs.python.org/3/library/asyncio-subprocess.html